pkg/proxy/proxy.go

```diff
@@ -90,78 +125,50 @@ func (p *Proxy) unlockVIN(vin string) {
 
 // New creates an http proxy.
 //
-// Vehicles must have the public part of skey enrolled on their keychains. (This is a
-// command-authentication key, not a TLS key.)
-func New(ctx context.Context, skey protocol.ECDHPrivateKey, cacheSize int) (*Proxy, error) {
-	return &Proxy{
-		Timeout:    DefaultTimeout,
-		commandKey: skey,
-		sessions:   cache.New(cacheSize),
-	}, nil
+// Vehicles must have the public part of skey enrolled on their keychains.
+// (This is a command-authentication key, not a TLS key.)
+func New(ctx context.Context, skey protocol.ECDHPrivateKey, cacheSize int, accountProvider AccountProvider) *Proxy {
```

Comment:
Consider returning an (always nil) error for the sake of API stability. This would also allow API stability going forward of the implementation changes in a way that re-introduces the possibility of failing.